Sourcery/main.swift

```diff
@@ -67,8 +67,22 @@ fileprivate enum Validators {
         return path
     }
 
-    static func isWriteable(path: Path) -> Path {
-        if path.exists && !path.isWritable {
+    static func isWritable(path: Path, autoLock: Bool) -> Path {
+        let pathExistsAndIsNotWritable = path.exists && !path.isWritable
+        let pathExistsAndWillNotBeWritableIfUnlocked: Bool = {
+            guard path.exists, autoLock, path.isImmutable else {
+                return false
+            }
+            
+            try! path.setImmutable(false)
```

Comment:
Is it correct if we try to right to the folder that is already locked and it was not Sourcery who locked it.